### PR TITLE
Adding the Actual functionality of the lib.

### PR DESCRIPTION
**Pull request by yasserhennawi**

### CODE REVIEW

File: src/HotTyper/test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import MovingCursor from './index';
+
+const WrapperStyle = {
```
Comment (bitriddler) [inline src/HotTyper/test.js]: why pascal case for this variable 🤔 ?

File: src/HotTyper/index.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    ]),
+    hideCursorOnEnd: PropTypes.bool,
+    speedOfLoop: PropTypes.number,
+    typingRate: PropTypes.number,
+    cursorFlashRate: PropTypes.number,
+    initialDelay: PropTypes.number
+  };
+
+  static defaultProps = {
+    text: [
```
Comment (bitriddler) [inline src/HotTyper/index.js]: I think we shouldn't have a default props for text

File: src/HotTyper/index.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  }
+
+  componentDidMount() {
+    setTimeout(
+      this.setState.bind(this, { isTyping: true }),
+      this.props.initialDelay
+    );
+  }
+
+  componentWillUpdate(nextProps, nextState) {
```
Comment (bitriddler) [inline src/HotTyper/index.js]: use `componentDidUpdate`
https://reactjs.org/docs/react-component.html#unsafe_componentwillupdate

File: src/HotTyper/test.js (hunk 1 above)
Comment (yasserhennawi) [reply src/HotTyper/test.js]: Sorry, Migrated this from a StyledComponent and forgot the naming 😅

**Comment by yasserhennawi** [comment]
The just-pushed commit `e3cff27` should:
  - Fix :
> Use componentDidUpdate instead of componentWillUpdate

  - Fix :
> `text` prop should be required (i.e no defaultProp for it)

  - Upgrade both `react` and `react-dom` dev-deps to 16.5.2 or newer minor level version.

@bitriddler If you think it's good enough for now, we can merge, bump to V1.0.0 and start on the demo page?

**Comment by yasserhennawi** [comment]
Merging this and let the party begin! 🎉🎉 🎈

**Pull request merged**